src/Glpi/Controller/GenericFormController.php

```diff
@@ -196,7 +193,7 @@ public function displayForm(CommonDBTM $object, Request $request): Response
         }
 
         return $this->render('pages/generic_form.html.twig', [
-            'id' => $request->query->get('id', -1),
+            'id' => $request->query->get('id', 0),
```

Comment:
Could you explain this change? I am not sure that it will have no side effect.

Reply:
It is strange how "invalid IDs" are handled internally. If I let this value to `-1`, the route `/front/certificate.form.php`/`/Certificate/Form` will not display the QR code for the "Asset URL" field. With the value "0", it displays it. I tried to search and didn't find why, but if you have a clue, maybe this could be fixed downstream instead of in the generic controller 👌

Reply:
Will be fixed by #18492

Reply:
Since the `getId()` method will return `-1`, I guess this kind of change can indeed have impact, depending on how tests are done in the underlying code... We should use `isNewItem` rather than compare the value of `fields ['id']` or `getId()` but we currently cannot assume things are done right :/

Reply:
Great, will remove this change from my PR 👌